spec/restful_model_spec.rb

```diff
@@ -58,12 +58,14 @@ def foo=(bar); end
 
     it "should pass parameters as query parameters when calling delete" do
       url = 'http://localhost:5555/events/1'
-      stubbed_url = 'http://localhost:5555/events/1?param1=&param2=stuff&send_notifications=true'
+      stubbed_url = /events\/1\?param1[=]?&param2=stuff&send_notifications=true/
```

Comment:
Hmm, could you give me a bit more details about the change you made to the test?

Reply:
Sure!

WebMock supports stubbing URLs in a number of ways, ranging from a straight string comparison, to a "with" syntax that describes the different parts of the URI, to a regular expression.

From my perspective, tests should focus on the thing they actually care about, in this case "Hey, when you send a parameter as nil, there isn't a value included in the query string" and not the incidentals ("Hey, we send all our requests to the right host, right?") that are taken care of in other parts.

I removed the bit of the stub/assertion that expects localhost:5555 (since that is incidental) and switched to regex syntax so that the library change where rest-client went from sending `param=` to `param` for nil params doesn't impact the test anymore.

My "ideal" API tests follow the arrange/act/assert format:

```
# arrange
# (Make stubs as un-greedy as possible)

# act
# fire off the actual request

# assert
# Make asserts about what query parameters got what value in the stubs here, and make asserts against the resulting object
```

I, personally, don't mind multiple asserts per test definition;  though if you have strong feels about that we'd probably want to look into something like `rspec-given` to make the Arrange/Act/Assert pattern more explicit while exposing each assert as it's own test definition.

I didn't want to make *too* broad of a change in this PR, however.

Reply:
Make sense, thanks!